addons/stock/models/stock_move.py

```diff
@@ -286,6 +287,12 @@ def _compute_product_qty(self):
             move.product_qty = move.product_uom._compute_quantity(
                 move.product_uom_qty, move.product_id.uom_id, rounding_method=rounding_method)
 
+    @api.depends('picking_partner_id')
+    def _compute_picking_partner_id(self):
+        for move in self:
+            if move in move.picking_id.move_ids_without_package:
+                move.partner_id = move.picking_partner_id
+
```

Comment:
This will break the dropshipping case :thinking: maybe an override in `purchase_stock` (since that's where the partner_id is manually written, https://github.com/odoo/odoo/blob/edaf37eeb9244976ecc6c60aec089f9a2b4e3020/addons/purchase_stock/models/purchase.py#L521
even though `stock.move.is_dropshipped` is in `stock_account`)

 I'm surprised there isn't a test for checking that the `partner_id` of a dropshipped move is correct :scream: 

steps to reproduce:
- install sale/purchase/stock + activate dropshipping setting
- create a product w/ dropship route + purchase vendor
- sell the product to a customer (NOT the vendor) + confirm the PO from the vendor
- add the `stock.move.partner_id` into the view of the picking form view
partner_id should = the customer, not the vendor

Also: 
```suggestion
    @api.depends('picking_id.partner_id')
    def _compute_picking_partner_id(self):
        for move in self:
            if move in move.picking_id.move_ids_without_package:
                move.partner_id = move.picking_id.partner_id
```
since `picking_partner_id` was removed in v15.

Also we should delete `onchange_partner_id` in stock_picking since it will be redundant with this... So perhaps we should do this fix in master and ignore the bug in stable for now unless someone complaints about it (but it's also not blocking so there is a workaround, so maybe we will never fix it)